source/Halibut/Transport/PollingClient.cs

```diff
@@ -16,25 +16,27 @@ public class PollingClient : IPollingClient
         readonly Thread thread;
         readonly CancellationToken cancellationToken;
         bool working;
+        Func<RetryPolicy> CreateRetryPolicy;
 
         [Obsolete("Use the overload that provides a logger. This remains for backwards compatibility.")]
-        public PollingClient(Uri subscription, ISecureClient secureClient, Func<RequestMessage, ResponseMessage> handleIncomingRequest)
-            : this(subscription, secureClient, handleIncomingRequest, null)
+        public PollingClient(Uri subscription, ISecureClient secureClient, Func<RequestMessage, ResponseMessage> handleIncomingRequest, Func<RetryPolicy> createRetryPolicy)
```

Comment:
Why `Func<RetryPolicy> createRetryPolicy` rather than just passing the RetryPolicy?

Reply:
Its mutable :(

Reply:
The retry policy is holding both the backoff exponent, min and max times as well as the stopwatches used to record how long since the last call occurred. If we passed the retry policy around then this same stopwatch would be shared by all connections to the client e.g. when a tentacle is configured to communicate to multiple octopus servers.

Reply:
😭 Ahh yeh!!